ssl/s3_lib.c

```diff
@@ -3714,8 +3714,10 @@ const SSL_CIPHER *ssl3_choose_cipher(SSL *s, STACK_OF(SSL_CIPHER) *clnt,
         allow = srvr;
     }
 
-    tls1_set_cert_validity(s);
-    ssl_set_masks(s);
+    if (!SSL_IS_TLS13(s)) {
+        tls1_set_cert_validity(s);
```

Comment:
tls1_set_cert_validity() calls tls1_check_chain() repeatedly for the various cert types. A lot of the checks in tls1_check_chain() only occur if suitb_flags is set, or strict_mode is set. These checks are only relevant to suiteb which is only well defined for TLSv1.2 (not TLSv1.3).

The only check that is actually relevant in this context to non-suiteb is the call to tls1_check_cert_param(). This only checks that the sig-algs and curve are appropriate to the certificate. In TLSv1.3 we do this instead in tls_choose_sigalg().